packages/components/src/utils/events.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * Merges event handlers together.
+ *
+ * @template TEvent
+ * @param {(event: TEvent) => void} handler
+ * @param {(event: TEvent) => void} otherHandler
+ */
+export function mergeEvent( handler, otherHandler ) {
+	return (
+		/* eslint-disable jsdoc/no-undefined-types */
+		/** @type {TEvent} */
+		/* eslint-enable jsdoc/no-undefined-types */
+		event
+	) => {
+		if ( typeof handler === 'function' ) {
+			handler( event );
+		}
+		if ( typeof otherHandler === 'function' ) {
+			otherHandler( event );
+		}
+	};
+}
+
+/**
+ * Merges a set of event handlers together.
+ *
+ * @template TEvent
+ * @param {Record<string, (event: TEvent) => void>} handlers
+ * @param {Record<string, (event: TEvent) => void>} extraHandlers
+ */
+export function mergeEventHandlers( handlers = {}, extraHandlers = {} ) {
+	const mergedHandlers = { ...handlers };
+
+	for ( const [ key, handler ] of Object.entries( mergedHandlers ) ) {
+		if ( typeof extraHandlers[ key ] === 'function' ) {
+			mergedHandlers[ key ] = mergeEvent( handler, extraHandlers[ key ] );
+		}
+	}
+
+	return mergedHandlers;
+}
```

Comment:
This seems like it should go into `compose` under a new `useMergedEventHandlers` hook. What do you think?

Reply:
I'll defer to your instincts here. 

Would it make sense to do these package migrations in a separate PR?